woocommerce/woocommerce-android
# Issue/4047 Remove checkstyle IDEA plugin. Updated checkstyle rules

Closes Issue #4047 

Don't merge before the target branch is merged

* Removes checkstyle IDEA plugin automatic installation
* Removes rules that don't work (java related or other projects)
* Removed suppression which not relevant anymore

### How to test
Check `checkstyle.xml` and it's regexp based rules.
Modify code locally to break one of the rules. Run `./gradlew checkstyle` and notice the violations report
